### DISCUSSION TITLE
YouTube Embeds -  Cannot find an embed for this URL.

### DISCUSSION THREAD

**Discussion by alexandremjacques** [wagtail/wagtail#10828] [General]
Hi,
I'm trying to create a simple page (HomePage) with YT videos embedded. So far I was able to add the fields in the admin but I'm not able to register the video URL with the error message `Cannot find an embed for this URL.`

`models.py`
```python
class HomePage(Page):
    max_count = 1
    subtitle = models.CharField(max_length=200, blank=True)

    videos = StreamField(
        [
            ("youtubevideos", blocks.YouTubeVideoBlock())
        ],
        null=True,
        blank=True,
        use_json_field=True
    )

    content_panels = Page.content_panels + [
        FieldPanel('subtitle'),
        FieldPanel('videos')
    ]
```
 
`blocks.py`
```python
class YouTubeVideoBlock(blocks.StructBlock):
    url = EmbedBlock(required=True)

    class Meta:
        icon = 'video'
        label = 'YouTube Video'
```

`base.py`
```python
WAGTAILEMBEDS_FINDERS = [
    {
        'class': 'wagtail.embeds.finders.oembed',
        'providers': [youtube],
    }
]
```

<img width="944" alt="image" src="https://github.com/wagtail/wagtail/assets/1953298/0f666760-9d02-4366-8281-8d95925d2ccc">

Any help is appreciated.

Cheers!

**Comment by gasman** [comment]
Assuming I've copied that URL correctly: https://www.youtube.com/watch?v=blwl8GpfbGQ - "This video isn't available anymore".

**Comment by alexandremjacques** [comment]
Any URL that I try, fails: https://www.youtube.com/watch?v=jfKfPfyJRdk

<img width="961" alt="image" src="https://github.com/wagtail/wagtail/assets/1953298/2eef57af-1b4d-46be-99c8-58eefb522ec8">

**Comment by alexandremjacques** [comment]
Any other suggestions? Kinda stuck. No error messages on the logs.

**Reply by zerolab** [reply]
I have just tried https://www.youtube.com/watch?v=jfKfPfyJRdk with Bakerydemo's embed block and it works. Are you sure your project doesn't change the embed finders
https://docs.wagtail.org/en/stable/advanced_topics/embeds.html#configuring-embed-finders or does other customisation?

**Reply by alexandremjacques** [reply]
What I shared here is pretty much all the code that I have. I saw something in a post about certificate error on macOS. Will try it (even though I don't see any errors in the logs).

Thanks.
